src/app/page.tsx

```diff
@@ -1,91 +1,72 @@
-"use client";
+import { AdvocateCard } from './AdvocateCard';
+import { DEFAULT_LIMIT, DEFAULT_PAGE } from './constants';
+import PageLimitSelect from './PageLimitSelect';
+import SearchForm from './SearchForm';
+import { advocateService } from './services/advocateService';
+import { Advocate } from './types/advocate';
 
-import { useEffect, useState } from "react";
+export default async function Home({
+  searchParams,
+}: {
+  searchParams: { [key: string]: string | string[] | undefined };
+}) {
+  const page = parseInt(searchParams.page as string) || DEFAULT_PAGE;
+  const limit = parseInt(searchParams.limit as string) || DEFAULT_LIMIT;
+  const search = (searchParams.search as string) || '';
 
-export default function Home() {
-  const [advocates, setAdvocates] = useState([]);
-  const [filteredAdvocates, setFilteredAdvocates] = useState([]);
+  const { data, rows } = await advocateService.getAdvocates({
+    page,
+    limit,
+    search,
+  });
 
-  useEffect(() => {
-    console.log("fetching advocates...");
-    fetch("/api/advocates").then((response) => {
-      response.json().then((jsonResponse) => {
-        setAdvocates(jsonResponse.data);
-        setFilteredAdvocates(jsonResponse.data);
-      });
-    });
-  }, []);
-
-  const onChange = (e) => {
-    const searchTerm = e.target.value;
-
-    document.getElementById("search-term").innerHTML = searchTerm;
-
-    console.log("filtering advocates...");
-    const filteredAdvocates = advocates.filter((advocate) => {
-      return (
-        advocate.firstName.includes(searchTerm) ||
-        advocate.lastName.includes(searchTerm) ||
-        advocate.city.includes(searchTerm) ||
-        advocate.degree.includes(searchTerm) ||
-        advocate.specialties.includes(searchTerm) ||
-        advocate.yearsOfExperience.includes(searchTerm)
-      );
-    });
-
-    setFilteredAdvocates(filteredAdvocates);
-  };
+  return (
+    <main className='w-full'>
+      <header>
+        <h2 className='h-12 bg-primary font-extrabold text-white flex justify-center items-center'>
+          Solace Advocate Finder
+        </h2>
+      </header>
+      <SearchForm />
+      <section aria-label='Advocates List'>
+        <div className='grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 p-4 bg-slate-50'>
+          {data.map((advocate: Advocate) => (
+            <AdvocateCard key={advocate.id} advocate={advocate} />
```

Comment:
Nit to myself, but I could have spread on advocate here, and altered the AdvocateCard interface to contain all the fields of the Advocate type, which would have cleaned up a lot of dot notation in that component.  Maybe even some extraction inside the callback params.  Something like 

```suggestion
            {data.map({id, ...rest}: Advocate) => (
               <AdvocateCard key={id} {...rest} /> 
            )}
```